File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/tilecache/ConvertTileCache.java
package com.valentingrigorean.arcgis_maps_flutter.tasks.tilecache;

import androidx.annotation.NonNull;

import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheParameters;
import com.valentingrigorean.arcgis_maps_flutter.Convert;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class ConvertTileCache extends Convert {

    @NonNull
    public static ExportTileCacheParameters toExportTileCacheParameters(Object o) {
        final Map<?, ?> data = toMap(o);
        final ExportTileCacheParameters parameters = new ExportTileCacheParameters();
        final Object areaOfInterest = data.get("areaOfInterest");
        if (areaOfInterest != null) {
            parameters.setAreaOfInterest(Convert.toGeometry(areaOfInterest));
        }
        parameters.setCompressionQuality(toFloat(data.get("compressionQuality")));
        parameters.getLevelIDs().addAll(toList(data.get("levelIds")).stream().map(i -> toInt(i)).collect(java.util.stream.Collectors.toList()));
        return parameters;
    }

    public static Object exportTileCacheParametersToJson(@NonNull ExportTileCacheParameters parameters) {
        final Map<String, Object> data = new HashMap<>(3);
        if (parameters.getAreaOfInterest() != null) {
            data.put("areaOfInterest", Convert.geometryToJson(parameters.getAreaOfInterest()));
        }
        data.put("compressionQuality", parameters.getCompressionQuality());
        data.put("levelIds", parameters.getLevelIDs());

        return data;
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/tilecache/ConvertTileCache.kt
package com.valentingrigorean.arcgis_maps_flutter.tasks.tilecache

import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheParameters
import com.valentingrigorean.arcgis_maps_flutter.Convert
import com.valentingrigorean.arcgis_maps_flutter.utils.toMap
import java.util.stream.Collectors

object ConvertTileCache : Convert() {
    fun toExportTileCacheParameters(o: Any?): ExportTileCacheParameters {
        val data: Map<*, *> = Convert.Companion.toMap(
            o!!
        )
        val parameters = ExportTileCacheParameters()
        val areaOfInterest = data["areaOfInterest"]
        if (areaOfInterest != null) {
            parameters.areaOfInterest = Convert.Companion.toGeometry(areaOfInterest)
        }
        parameters.compressionQuality = Convert.Companion.toFloat(
            data["compressionQuality"]
        )
        parameters.levelIDs.addAll(Convert.Companion.toList(
            data["levelIds"]!!
        ).stream().map<Int> { i: Any? -> Convert.Companion.toInt(i) }
            .collect<List<Int>, Any>(Collectors.toList<Int>()))
        return parameters
    }

    fun exportTileCacheParametersToJson(parameters: ExportTileCacheParameters): Any {
        val data: MutableMap<String, Any?> = HashMap(3)
        if (parameters.areaOfInterest != null) {
            data["areaOfInterest"] =
                Convert.Companion.geometryToJson(parameters.areaOfInterest)
        }
        data["compressionQuality"] = parameters.compressionQuality
        data["levelIds"] = parameters.levelIDs
        return data
    }
}